src/render/draw_custom.js

```diff
@@ -0,0 +1,58 @@
+// @flow
+
+export default drawCustom;
+
+import DepthMode from '../gl/depth_mode';
+import StencilMode from '../gl/stencil_mode';
+import {prepareOffscreenFramebuffer, drawOffscreenTexture} from './offscreen';
+
+import type Painter from './painter';
+import type SourceCache from '../source/source_cache';
+import type CustomStyleLayer from '../style/style_layer/custom_style_layer';
+
+function drawCustom(painter: Painter, sourceCache: SourceCache, layer: CustomStyleLayer) {
+
+    const context = painter.context;
+    const implementation = layer.implementation;
+
+    if (painter.renderPass === 'offscreen') {
+
+        const prerender = implementation.prerender;
+        if (prerender) {
+            painter.setCustomLayerDefaults();
+
+            prerender.call(implementation, context.gl, painter.transform.customLayerMatrix());
```

Comment:
Nit: no need for `call` here, `implementation.prerender(...)`

Reply:
This was done to satisfy flow complaining about `implementation.prerender` being potentially undefined. The alternative is a cast through `any`. Would this be better?

Reply:
@ansis this is surprising — Flow should detect the check in `if (implementation.prerender)` and not complain in this case.

Reply:
https://flow.org/en/docs/lang/refinements/#toc-refinement-invalidations